rest/model/codegen.go

```diff
@@ -395,10 +427,116 @@ func generateServiceConversions(structVal *types.Struct, packageName, structName
 	return output(serviceTemplate, data)
 }
 
+// generateArrayConversions does a similar task to generateServiceConversions above, but specifically
+// generates code which will convert arrays whose elements are arbitrary types. The approach
+// here has logical similarities to generating conversion code for both a struct and a scalar
+func generateArrayConversions(modelName string, modelType types.Type, restName, restType string, outIsPtr bool, generatedConversions map[typeInfo]string) (string, error) {
+	arrayElem := findArrayElem(modelType)
+	if arrayElem == nil {
+		return "", errors.Errorf("unexpected error: %s is not an array", modelName)
+	}
+	conversionInfo := typeInfo{
+		InType:  modelType.String(),
+		OutType: "temp",
+	}
+	if generatedConversions[conversionInfo] != "" {
+		return "", nil
+	}
+
+	arrayTemplate, err := getTemplate(arrayMethodsTemplatePath)
+	if err != nil {
+		return "", errors.Wrap(err, "error getting service methods template")
+	}
+	lineTemplate, err := getTemplate(funcTemplatePath)
+	if err != nil {
+		return "", errors.Wrap(err, "error getting line template")
+	}
+	opts := convertFnOpts{
+		in:       modelType,
+		outIsPtr: outIsPtr,
+		outType:  restType,
+	}
+	convertFuncs, err := arrayConversionFn(opts, generatedConversions)
+	if err != nil {
+		return "", err
+	}
+	bfsData := conversionLine{
+		TypeConversionFunc: convertFuncs.converter,
+	}
+	bfsCode, err := output(lineTemplate, bfsData)
+	if err != nil {
+		return "", errors.Wrap(err, "error generating line code")
+	}
+	tsData := conversionLine{
+		TypeConversionFunc: convertFuncs.inverter,
+	}
+	tsCode, err := output(lineTemplate, tsData)
+	if err != nil {
+		return "", errors.Wrap(err, "error generating line code")
+	}
+	modelTypeStr := fmt.Sprintf("[]%s", (*arrayElem).String())
+	serviceData := arrayConversionInfo{
+		FromType:       shortenPackage(modelTypeStr),
+		FromHasPtr:     containsPtr(modelTypeStr),
+		ToType:         shortenPackage(restType),
+		ToHasPtr:       containsPtr(restType),
+		ConversionCode: bfsCode,
+	}
+	code, err := output(arrayTemplate, serviceData)
+	if err != nil {
+		return "", err
+	}
+	if modelTypeStr != restType {
```

Comment:
Even in the case that these _do_ equal each other, is it a problem that we don't have an arrayConversionCode with the tsCode Conversion? Or do we only need tsCode in this case? Because then 473 can be moved into here.